controllers/errors.go

```diff
@@ -0,0 +1,14 @@
+package controllers
+
+import (
+	"fmt"
+)
+
+type UpgradeLimitExceededError struct {
+	NodeName string
+	Limit    int
```

Comment:
The limit is constant and can be accessed from MaxParallelUpgrades. What may be relevant here is the `count`, the actual number of upgrading nodes.